AutomatedTesting/Gem/PythonTests/prefab/Prefab_Operations.py

```diff
@@ -0,0 +1,165 @@
+    instances_removed_expected_name_counts = Counter()
+
+    entity_ids_to_remove = list(container_entity_ids)
+    while entity_ids_to_remove:
+        entity_id = entity_ids_to_remove.pop(-1)
+        entity_name = editor.EditorEntityInfoRequestBus(bus.Event, 'GetName', entity_id)
+        instances_to_remove_name_counts[entity_name] += 1
+
+        children_entity_ids = editor.EditorEntityInfoRequestBus(bus.Event, 'GetChildren', entity_id)
+        for child_entity_id in children_entity_ids:
+            entity_ids_to_remove.append(child_entity_id)
+
+    for entity_name, entity_count in instances_to_remove_name_counts.items():
+        entities = prefab_test_utils.find_entities_by_name(entity_name)
+        instances_removed_expected_name_counts[entity_name] = len(entities) - entity_count
+
+    delete_prefab_result = prefab.PrefabPublicRequestBus(bus.Broadcast, 'DeleteEntitiesAndAllDescendantsInInstance', container_entity_ids)
+    if delete_prefab_result.IsSuccess():
+        print(Results.delete_prefab_succeeded)
+    else:
+        print(Results.delete_prefab_failed)
+        return
+
+    prefab_test_utils.wait_for_propagation()
+
+    prefab_entities_deleted = True
+    for entity_name, expected_entity_count in instances_removed_expected_name_counts.items():
+        actual_entity_count = len(prefab_test_utils.find_entities_by_name(entity_name))
+        if actual_entity_count is not expected_entity_count:
+            prefab_entities_deleted = False
+            break
+
+    if prefab_entities_deleted:
+        print(Results.delete_prefab_entities_and_descendants_succeeded)
+    else:
+        print(Results.delete_prefab_entities_and_descendants_failed)
+
+
+async def reparent_prefab(container_entity_id, parent_entity_id):
+    prefab_name = editor.EditorEntityInfoRequestBus(bus.Event, 'GetName', container_entity_id)
+    current_children_entity_ids_having_prefab_name = prefab_test_utils.get_children_by_name(parent_entity_id, prefab_name)
+
+    pyside_utils.run_soon(lambda: editor.EditorEntityAPIBus(bus.Event, 'SetParent', container_entity_id, parent_entity_id))
+    pyside_utils.run_soon(lambda: prefab_test_utils.wait_for_propagation())
+
+    try:
+        active_modal_widget = await pyside_utils.wait_for_modal_widget()
+        error_message_box = active_modal_widget.findChild(QtWidgets.QMessageBox)
+        ok_button = error_message_box.button(QtWidgets.QMessageBox.Ok)
+        ok_button.click()
+        print(Results.reparent_prefab_cyclical_dependency_detected)
+        return EntityId()
+    except pyside_utils.EventLoopTimeoutException:
+        pass        
+
+    updated_children_entity_ids_having_prefab_name = prefab_test_utils.get_children_by_name(parent_entity_id, prefab_name)
+    if len(updated_children_entity_ids_having_prefab_name) is not len(current_children_entity_ids_having_prefab_name) + 1:
+        print(Results.reparent_prefab_parent_not_own_prefab)
+        return EntityId()
```

Comment:
Fetching entities by name just to do validation is getting messier when trying to understand what the test does. This would make iterating over tests very difficult. I suggest prioritize fixing this bug https://github.com/o3de/o3de/issues/3555 and refactoring these tests after this PR gets submitted

Reply:
So are you suggesting to remove this now? Or fix this part after fixing the bug?

Reply:
Created a ticket for the refactoring: https://jira.agscollab.com/browse/LYN-5831